src/scratch.h

```diff
@@ -0,0 +1,35 @@
+/**********************************************************************
+ * Copyright (c) 2017 Andrew Poelstra	                              *
+ * Distributed under the MIT software license, see the accompanying   *
+ * file COPYING or http://www.opensource.org/licenses/mit-license.php.*
+ **********************************************************************/
+
+#ifndef _SECP256K1_SCRATCH_
+#define _SECP256K1_SCRATCH_
+#include <sys/types.h>
```

Comment:
We can remove this #include. There are two places that `ssize_t` are used in `ecmult_impl.h` which need to be removed. The first, `ssize_t max_alloc =` can be changed to `size_t` because it is assigned a `size_t` anyway (we changed `secp256k1_scratch_max_allocation` to not return a sign value).

The second use is `ssize_t space_for_points`. This can also be made into a `size_t`. The check on line 938, `if (space_for_points < 0)`, will need to be rearranged to compare the two things that are subtracted on line 937 to get `space_for_points`.